packages/middleware-sdk-s3/src/validate-bucket-name.ts

```diff
@@ -7,16 +7,19 @@ import {
   MetadataBearer,
   Pluggable,
 } from "@aws-sdk/types";
+import { validate } from "@aws-sdk/util-arn-parser";
 
 export function validateBucketNameMiddleware(): InitializeMiddleware<any, any> {
   return <Output extends MetadataBearer>(
     next: InitializeHandler<any, Output>
   ): InitializeHandler<any, Output> => async (
     args: InitializeHandlerArguments<any>
   ): Promise<InitializeHandlerOutput<Output>> => {
-    const { input } = args;
-    if (typeof input.Bucket === "string" && input.Bucket.indexOf("/") >= 0) {
-      const err = new Error(`Bucket name shouldn't contain '/', received '${input.Bucket}'`);
+    const {
+      input: { Bucket },
+    } = args;
+    if (typeof Bucket === "string" && !validate(Bucket) && Bucket.indexOf("/") >= 0) {
```

Comment:
If importing as validateArn on line 9
```suggestion
    if (typeof Bucket === "string" && !validateArn(Bucket) && Bucket.indexOf("/") >= 0) {
```